### PR TITLE
Feat: Get raw storage file (issue #60)

### PR DESCRIPTION
**Pull request by MHHukiewitz**
Problem: There's no function to download storage data by its item hash (see issue #60)

Solution: Add get_store_file()

### CODE REVIEW

File: src/aleph_client/asynchronous.py [hunk 1]
```diff
@@ -562,6 +562,36 @@ async def get_posts(
...
+    :param storage_engine: The storage engine to use. (DEFAULT: "storage")
+    :param session: The aiohttp session to use. (DEFAULT: get_fallback_session())
+    :param api_server: The API server to use. (DEFAULT: "https://api2.aleph.im")
+    """
+    storage_engine = storage_engine or StorageEnum.storage
+    session = session or get_fallback_session()
+    api_server = api_server or settings.API_HOST
+
+    if storage_engine == StorageEnum.storage:
+        async with session.get(f"{api_server}/api/v0/storage/raw/{file_hash}") as response:
```
Comment (odesenfans) [inline src/aleph_client/asynchronous.py:584]: Independently of the storage engine, you can always use `/api/v0/storage/raw/{file_hash}`. There's no `/api/v0/ipfs/files/read/{file_hash}` endpoint afaik.

File: src/aleph_client/asynchronous.py [hunk 2]
```diff
@@ -562,6 +562,36 @@ async def get_posts(
         return await resp.json()
 
 
+async def get_store_file(
```
Comment (odesenfans) [inline src/aleph_client/asynchronous.py:565]: `download_file` ?

File: src/aleph_client/asynchronous.py [hunk 3]
```diff
@@ -562,6 +562,36 @@ async def get_posts(
...
+        file_hash: str,
+        storage_engine: Optional[StorageEnum] = None,
+        session: Optional[ClientSession] = None,
+        api_server: Optional[str] = None
+) -> bytes:
```
Comment (odesenfans) [inline src/aleph_client/asynchronous.py:570]: Beware of large files, this function might end up using a lot of memory.

Comment (MHHukiewitz) [reply src/aleph_client/asynchronous.py:569]: Recommend some type of generator then? Added a warning for starters ^^'

Comment (odesenfans) [reply src/aleph_client/asynchronous.py:569]: A warning is a good start. A bytes generator might be a good idea, as long as what the user is doing with it does not take too long. And then adding higher-level functions on top of the generator, like `def download_file(file_hash: str, output_file: pathlib.Path)` etc.

Comment (MHHukiewitz) [reply src/aleph_client/asynchronous.py:569]: Judging from https://github.com/aleph-im/pyaleph/blob/190ebef84b5823c4ea5616f5152b4f6fa21b17b9/src/aleph/web/controllers/storage.py#L79
there is currently no way to request a StreamingResponse, so we inadvertently will have the whole file in memory until we store/dispose of it in a file, correct?

Comment (hoh) [reply src/aleph_client/asynchronous.py:569]: Indeed, but that is on the server side with lots of RAM.

**Review by odesenfans — APPROVED** [review]

File: src/aleph_client/asynchronous.py (hunk 3 above)
Comment (hoh) [reply src/aleph_client/asynchronous.py:569]: Returning a file descriptor may be ideal. 

Headers contain the size of the content, which could be stored in a temporary file when larger than a certain threshold.

Comment (hoh) [reply src/aleph_client/asynchronous.py:569]: See https://docs.python.org/3/library/io.html#io.BufferedReader

Comment (MHHukiewitz) [reply src/aleph_client/asynchronous.py:569]: Let's create an issue for this then :+1:

**Pull request merged**